src/main/java/bio/terra/cli/service/GoogleOauth.java

```diff
@@ -252,19 +405,40 @@ public static ServiceAccountCredentials getServiceAccountCredential(
   /**
    * Refresh the credential if expired and then return its access token.
    *
-   * @param credential credentials object
+   * @param credential Terra credentials object
    * @return access token
    */
-  public static AccessToken getAccessToken(GoogleCredentials credential) {
+  public static AccessToken getAccessToken(TerraCredentials credential) {
     try {
-      credential.refreshIfExpired();
+      credential.getGoogleCredentials().refreshIfExpired();
     } catch (IOException ioEx) {
       logger.warn("Error refreshing access token", ioEx);
       // don't throw an exception here because the token may not be expired, in which case it's fine
```

Comment:
I don't fully get the intention either; speculatively, `refreshIfExpired()` is supposed to refresh "if expired or expiring imminently", so I guess maybe the plan is to try the call anyway in case it was "imminent" and refresh call hit a spurious error.  Then maybe the downstream call would succeed due to the token not yet being expired (and/or clock skew)?

Reply:
Seems like a dubious optimization, but it's plausible. Thanks.